plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java

```diff
@@ -0,0 +1,662 @@
+        String entity = "{\n"
+            + "  \"name\": \"OpenAI moderations model Connector\",\n"
+            + "  \"description\": \"The connector to public OpenAI moderations model service\",\n"
+            + "  \"version\": 1,\n"
+            + "  \"protocol\": \"http\",\n"
+            + "  \"parameters\": {\n"
+            + "      \"endpoint\": \"api.openai.com\",\n"
+            + "      \"auth\": \"API_Key\",\n"
+            + "      \"content_type\": \"application/json\",\n"
+            + "      \"model\": \"moderations\"\n"
+            + "  },\n"
+            + "  \"credential\": {\n"
+            + "      \"openAI_key\": \""
+            + System.getenv("OPENAI_KEY")
+            + "\"\n"
+            + "  },\n"
+            + "  \"actions\": [\n"
+            + "      {\n"
+            + "      \"action_type\": \"predict\",\n"
+            + "          \"method\": \"POST\",\n"
+            + "          \"url\": \"https://api.openai.com/v1/moderations\",\n"
+            + "          \"headers\": { \n"
+            + "          \"Authorization\": \"Bearer ${credential.openAI_key}\"\n"
+            + "          },\n"
+            + "          \"request_body\": \"{ \\\"input\\\": \\\"${parameters.input}\\\" }\"\n"
+            + "      }\n"
+            + "  ]\n"
+            + "}";
+        Response response = createConnector(entity);
+        Map responseMap = parseResponseToMap(response);
+        String connectorId = (String) responseMap.get("connector_id");
+        response = registerRemoteModel("openAI-GPT-3.5 moderations model", connectorId);
+        responseMap = parseResponseToMap(response);
+        String taskId = (String) responseMap.get("task_id");
+        waitForTask(taskId, MLTaskState.COMPLETED);
+        response = getTask(taskId);
+        responseMap = parseResponseToMap(response);
+        String modelId = (String) responseMap.get("model_id");
+        response = deployRemoteModel(modelId);
+        responseMap = parseResponseToMap(response);
+        taskId = (String) responseMap.get("task_id");
+        waitForTask(taskId, MLTaskState.COMPLETED);
+        String predictInput = "{\n" + "  \"parameters\": {\n" + "      \"input\": \"I want to kill them.\"\n" + "  }\n" + "}";
+        response = predictRemoteModel(modelId, predictInput);
+        responseMap = parseResponseToMap(response);
+        List responseList = (List) responseMap.get("inference_results");
+        responseMap = (Map) responseList.get(0);
+        responseList = (List) responseMap.get("output");
+        responseMap = (Map) responseList.get(0);
+        responseMap = (Map) responseMap.get("dataAsMap");
+        responseList = (List) responseMap.get("results");
+        responseMap = (Map) responseList.get(0);
+        assertTrue((Boolean) responseMap.get("flagged"));
+        responseMap = (Map) responseMap.get("categories");
+        assertTrue((Boolean) responseMap.get("violence"));
+    }
+
+    @Ignore
```

Comment:
If the problem is still from the post-processing function, can you create a ticket describing this issue? We should fix this problem caught in this test.

Reply:
Do you mean a github issue?

Reply:
Yes.

Reply:
Sure I'll do that